libgdf/src/gdf_table.cuh

```diff
@@ -586,10 +527,21 @@ public:
     // If either row contains a NULL, then by definition, because NULL != x for all x,
     // the two rows are not equal
     bool valid = this->is_row_valid(my_row_index) && other.is_row_valid(other_row_index);
-    if (false == valid) {
+    if (false == valid) 
+    {
       return false;
     }
 
+    // Generic lambda to check if two elements are equal between two columns
+    auto elements_are_equal = [](auto dispatched_type_var, void const * my_column, size_type my_row_index,
+                                 void const * other_column, size_type other_row_index)
+    {
+      using col_type = decltype(dispatched_type_var);
+      col_type const my_elem = static_cast<col_type const*>(my_column)[my_row_index];
+      col_type const other_elem = static_cast<col_type const*>(other_column)[other_row_index];
+      return my_elem == other_elem;
+    };
+
     for(size_type i = 0; i < num_columns; ++i)
```

Comment:
Note that this is a reimplementation of `std::all_of`. But never mind, this PR shouldn't fix everything...

Reply:
`std::all_of` doesn't work in device code.